lib/superagent-mock.js

```diff
@@ -142,7 +142,14 @@ function mock (superagent, config, logger) {
       }
     }
 
-    fn(error, response);
+    if (typeof context.delay === 'number') {
+      setTimeout(function () {
+        fn(error, response);
```

Comment:
`var response = parserMethod(match, fixtures);` instead of `response = parserMethod(match, fixtures);` and you remove the declaration of `response` above